### PR TITLE
EN-4623 : add more NTP servers for syncTime

### PR DESCRIPTION
**Pull request by miiu96**
Add more NTP servers for sync time.

### CODE REVIEW

File: ntp/syncTime.go [hunk 1]
```diff
@@ -76,14 +79,34 @@ func NewSyncTime(ntpConfig config.NTPConfig, syncPeriod time.Duration, customQue
...
 		syncPeriod:  syncPeriod,
 		query:       queryFunc,
-		ntpOptions:  NewNTPOptions(ntpConfig)}
+		ntpOptions:  NewNTPOptions(ntpConfig, hostIndex)}
+
 	return &s
 }
 
+func checkNTPHost(ntpConfig config.NTPConfig, customQueryFunc func(options NTPOptions) (*ntp.Response, error)) int {
+	ntpConfig.Timeout = 10 * time.Millisecond
```
Comment (SebastianMarian) [inline ntp/syncTime.go]: It is not too short for a timeout with only 10ms?

File: ntp/syncTime.go [hunk 2]
```diff
@@ -105,6 +128,13 @@ func (s *syncTime) sync() {
 
 			if err != nil {
 				log.Error(fmt.Sprintf("NTP Error: %s", err))
+				//Change host if current host return an error
```
Comment (SebastianMarian) [inline ntp/syncTime.go]: I would replace/remove now the constant totalRequests with len(s.ntpOptions.Host) in line 126. We might have maybe more than 10 ntp servers added in config and we should give a chance to anyone of them not only to the first 10(totalRequests).

File: ntp/syncTime.go [hunk 3]
```diff
@@ -17,32 +17,35 @@ var log = logger.DefaultLogger()
...
+	HostIndex    int
 }
 
 // NewNTPGoogleConfig creates an NTPConfig object that configures NTP to use
 // time.google.com. This is a useful default for tests, for example, to avoid
 // loading a configuration file just to have an NTPConfig.
 func NewNTPGoogleConfig() config.NTPConfig {
 	return config.NTPConfig{
-		Host:    "time.google.com",
+		Host:    []string{"time.google.com"},
```
[the remark this answers is not in the document]
Comment (miiu96) [reply ntp/syncTime.go]: done.

File: ntp/syncTime.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (miiu96) [reply ntp/syncTime.go]: done.

File: ntp/syncTime.go (hunk 1 above)
Comment (miiu96) [reply ntp/syncTime.go]: You right. Changed , and now the timeout used is the one from config file.

File: ntp/syncTime.go (hunk 2 above)
Comment (miiu96) [reply ntp/syncTime.go]: Done.

File: ntp/syncTime.go [hunk 4]
```diff
@@ -30,20 +27,25 @@ type NTPOptions struct {
...
 // NewNTPOptions creates a new NTPOptions object.
 func NewNTPOptions(ntpConfig config.NTPConfig, hostIndex int) NTPOptions {
+	if ntpConfig.TimeoutMilliseconds <= 0 {
+		ntpConfig.TimeoutMilliseconds = 100
+	}
+	timeout := time.Duration(ntpConfig.TimeoutMilliseconds) * time.Millisecond
+
 	return NTPOptions{
-		Host:         ntpConfig.Host,
+		Host:         ntpConfig.Hosts,
```
Comment (bogdan-rosianu) [inline ntp/syncTime.go]: `Hosts` also here

File: ntp/syncTime.go [hunk 5]
```diff
@@ -93,14 +94,12 @@ func NewSyncTime(ntpConfig config.NTPConfig, syncPeriod time.Duration, customQue
...
-	ntpConfig.Timeout = 10 * time.Millisecond
-
-	for hostIndex := range ntpConfig.Host {
+	for hostIndex := range ntpConfig.Hosts {
 		_, err := customQueryFunc(NewNTPOptions(ntpConfig, hostIndex))
 		if err != nil {
 			continue
 		}
-
+		log.Info(fmt.Sprintf("Using NTP server : %s", ntpConfig.Hosts[hostIndex]))
```
Comment (bogdan-rosianu) [inline ntp/syncTime.go]: use lowercase in the beginning of the log message

File: ntp/syncTime.go [hunk 6]
```diff
@@ -119,35 +118,33 @@ func (s *syncTime) StartSync() {
...
+			}, s.query)
+			s.ntpOptions.HostIndex = newHostIndex
 
-				continue
-			}
+			continue
+		}
 
-			log.Debug(fmt.Sprintf("NTP reading: %s", r.Time.Format("Mon Jan 2 15:04:05 MST 2006")))
+		log.Debug(fmt.Sprintf("NTP reading: %s", r.Time.Format("Mon Jan 2 15:04:05 MST 2006")))
```
Comment (bogdan-rosianu) [inline ntp/syncTime.go]: also here, lowercase log message

File: ntp/syncTime.go (hunk 4 above)
Comment (miiu96) [reply ntp/syncTime.go]: Done.

File: ntp/syncTime.go (hunk 5 above)
Comment (miiu96) [reply ntp/syncTime.go]: done.

File: ntp/syncTime.go (hunk 6 above)
Comment (miiu96) [reply ntp/syncTime.go]: in all the cases are used NPT with upper letters, i will not change it.

**Pull request merged**